server/punishments.js

```diff
@@ -1559,6 +1559,45 @@ Punishments.getRoomPunishments = function (user, options) {
 	return punishments;
 };
 
+/** @param {Room} room */
+Punishments.getPunishmentsOfRoom = function (room) {
+	let output = [];
+	const store = new Map();
+
+	if (Punishments.roomUserids.get(room.id)) {
+		for (let [key, value] of Punishments.roomUserids.get(room.id)) {
+			if (!store.has(value)) store.set(value, [new Set([]), new Set()]);
+			store.get(value)[0].add(key);
+		}
+	}
+
+	if (Punishments.roomIps.get(room.id)) {
+		for (let [key, value] of Punishments.roomIps.get(room.id)) {
+			if (!store.has(value)) store.set(value, [new Set([]), new Set()]);
+			store.get(value)[1].add(key);
+		}
+	}
+
+	if (store) {
```

Comment:
This is always true btw, you literally initialize the variable a few lines before. You can just iterate over it without this check.